File: backend/app/api/v1/endpoints/meta.py
from fastapi import APIRouter
from app.models.enums import FinancialCategory, StockUnit, MovementType, FinancialType

router = APIRouter(prefix="/meta", tags=["meta"])


@router.get("/enums")
def get_enums():
    return {
        "financial_categories": [e.value for e in FinancialCategory],
        "stock_units": [e.value for e in StockUnit],
        "movement_types": [e.value for e in MovementType],
        "financial_types": [e.value for e in FinancialType],
    }

File: frontend/src/services/meta.ts
import api from './api';

export interface MetaEnums {
  financial_categories: string[];
  stock_units: string[];
  movement_types: string[];
  financial_types: string[];
}

export const getEnums = async (): Promise<MetaEnums> => {
  const res = await api.get<MetaEnums>('/meta/enums');
  return res.data;
};

